src/Microsoft.ML.Core/Prediction/TrainContext.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime.Data;
+
+namespace Microsoft.ML.Runtime
+{
+    /// <summary>
+    /// Instances of this class are meant to be constructed and passed to trainers.
+    /// </summary>
+    public sealed class TrainContext
+    {
+        /// <summary>
+        /// The training set. Cannot be <c>null</c>.
+        /// </summary>
+        public RoleMappedData Train { get; }
+
+        /// <summary>
+        /// The validation set. Can be <c>null</c>.
+        /// </summary>
+        public RoleMappedData Validation { get; }
+
+        /// <summary>
+        /// The initial 
+        /// </summary>
+        public IPredictor InitialPredictor { get; }
```

Comment:
The initial predictor is no longer typed and we cannot through `ITrainer` alone force the trainer to identify what it is typed as. An optional interface on trainers for this specifically may be a superior solution. This hypothetical `IIncrementalTrainer<in TPred>` would *not* have a train method, but would be more or less a marker to indicate that a trainer might be able to incrementally train on something. Probably with a `bool CanIncrementallyTrain(TPred initPredictor)` method?

Unfortunately we may have to wait for C# 8